src/System.Text.Json/src/System/Text/Json/Serialization/Converters/JsonConverterEnum.cs

```diff
@@ -17,7 +17,7 @@ public override bool CanConvert(Type type)
             return type.IsEnum;
         }
 
-        protected override JsonConverter CreateConverter(Type type)
+        public override JsonConverter CreateConverter(Type type, JsonSerializerOptions options)
```

Comment:
In all of these cases where we're now taking options, what's the intended behavior if options is null?

Also, where are we actually utilizing the information from the options?

Reply:
It will never be null as the serializer picks up default options when you specify a null. We aren't actually utilizing the options yet, it was requested from API review to provide flexibility for authors without introducing a breaking change post 3.0.

Reply:
How can you guarantee it'll never be null? You're making it public.

Reply:
Well, from the serializer they won't be null.  But you're right, users can do it. The same holds true for the Read/Write methods on the converters, even though the classes are currently internal the methods are public and they can be obtained by the `GetConverter` method on the options. It would be up to the individual implementers to decide how to handle null if they use the options.

None of the current converters touch options. I didn't see any tests of this nature however and nothing actually checking what you get from GetConverter with a default options. I'll create an issue to track testing the GetConverter more thoroughly.

Reply:
#39457